### DISCUSSION TITLE
Using PEP-695 Generic TypeAliasType with Annotated Depends

### DISCUSSION THREAD

**Discussion by AlexanderPodorov** [fastapi/fastapi#11432] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
from fastapi import Depends

type Inject[T] = Annotated[T, Depends()]
```

### Description

See https://github.com/tiangolo/fastapi/discussions/9450 for the context.
Essentially the same use case, but it would be nicer to have PEP 695 style for that.

### Operating System

macOS

### FastAPI Version

0.110.1

### Pydantic Version

2.7.0

### Python Version

Python 3.12.3

**Comment by AlexanderPodorov** [comment]
Here is the minimal example, `fastapi==0.110.2`:
```python
from dataclasses import dataclass
from typing import Annotated, TypeVar
from uuid import UUID

from fastapi import Depends, FastAPI, Path


@dataclass
class SomePath:
    some_id: Annotated[UUID, Path()]


type Inject[T] = Annotated[T, Depends()]


_T = TypeVar("_T")


ClassicInject = Annotated[_T, Depends()]

app = FastAPI()


@app.get("/test/{some_id}")
def test(dep: Inject[SomePath]) -> UUID:
    return dep.some_id


@app.get("/test-classic/{some_id}")
def test_classic(dep: ClassicInject[SomePath]) -> UUID:
    return dep.some_id

```
Both endpoints should produce the same path parameter, but they are not.
`test` produces this one (incorrect):
![image](https://github.com/tiangolo/fastapi/assets/54511144/c5e05079-3f2a-4aaa-b227-6ddc2e703240)
`test-classic` produces this one (correct):
![image](https://github.com/tiangolo/fastapi/assets/54511144/70c5ec45-6784-444b-90b3-9874d66010da)

**Comment by YuriiMotov** [comment]
**Issue to track this:** https://github.com/fastapi/fastapi/issues/10719
